mithril-signer/src/message_adapters/from_pending_certificate_message.rs

```diff
@@ -16,7 +16,9 @@ fn to_signers(messages: &[SignerMessage]) -> StdResult<Vec<Signer>> {
             msg.party_id.to_owned(),
             ProtocolSignerVerificationKey::from_json_hex(&msg.verification_key)?,
             msg.verification_key_signature.to_owned(),
-            msg.operational_certificate.to_owned(),
+            msg.operational_certificate
+                .as_ref()
+                .and_then(|o| o.to_owned().try_into().ok()),
```

Comment:
Using .ok() hide eventual errors, you should propagate the error instead using a ?. Sadly you can't use ? inside closures so you will need to convert this to a match.
Also you should be consistent with the `ProtocolSignerVerificationKey` decoding and use `ProtocolOpCert::from_json_hex`.